### PR TITLE
ocamlPackages: Init ppx_cstubs at 0.6.1.1, add ppxlib 0.22.0, add ppx_deriving 5.2.1

### PR DESCRIPTION
**Pull request by ozanvos**
###### Motivation for this change

Add new package [ppx_cstubs](https://github.com/fdopen/ppx_cstubs/), update its dependency ppxlib, and update ppx_deriving

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
Please format the commit messages in style ``ocamlPackages.ppx_cstubs : Init at 0.6.1.1``

File: pkgs/development/ocaml-modules/ppx_cstubs/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+, re
+}:
+
+buildDunePackage rec {
+  pname = "ppx_cstubs";
+  version = "0.6.1.1";
+
+  useDune2 = true;
+
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ppx_cstubs/default.nix:20]: Please use fetchFromGitHub.

File: pkgs/development/ocaml-modules/ppx_cstubs/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  buildInputs = [
+    bigarray-compat
+    containers
+    cppo
+    ctypes
+    integers
+    num
+    (ppxlib.override { version = "0.22.0"; })
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ppx_cstubs/default.nix:32]: Please move the override to pkgs/top-level/ocaml-packages.nix.

File: pkgs/development/ocaml-modules/ppx_cstubs/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    re
+  ];
+
+  meta = with lib;
+    {
+      homepage = "https://github.com/fdopen/ppx_cstubs";
+      description = "Preprocessor for easier stub generation with ocaml-ctypes";
+      license = licenses.mit;
+      maintainers = [ maintainers.osener ];
+    };
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ppx_cstubs/default.nix:42]:
```suggestion
  meta = with lib; {
    homepage = "https://github.com/fdopen/ppx_cstubs";
    description = "Preprocessor for easier stub generation with ocaml-ctypes";
    license = licenses.mit;
    maintainers = [ maintainers.osener ];
  };
```

File: pkgs/development/ocaml-modules/ppx_deriving/default.nix [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
+, ocaml-migrate-parsetree
+, ocaml-migrate-parsetree-2-1
 }:
 
 let params =
-  if lib.versionAtLeast ppxlib.version "0.15"
-  then {
+  if lib.versionAtLeast ppxlib.version "0.20" then {
+    version = "5.2.1";
+    sha256 = "11h75dsbv3rs03pl67hdd3lbim7wjzh257ij9c75fcknbfr5ysz9";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ppx_deriving/default.nix:16]:
```suggestion
    sha256 = "11h75dsbv3rs03pl67hdd3lbim7wjzh257ij9c75fcknbfr5ysz9";
    useOMP2 = true;
```

File: pkgs/development/ocaml-modules/ppx_deriving/default.nix [hunk 5]
```diff
@@ -25,7 +37,13 @@ buildDunePackage rec {
   };
 
   buildInputs = [ ppxlib cppo ];
-  propagatedBuildInputs = [ ocaml-migrate-parsetree ppx_derivers result ];
+  propagatedBuildInputs = [
+    (if params.useOMP2 or true
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ppx_deriving/default.nix:41]:
```suggestion
    (if params.useOMP2
```

File: pkgs/top-level/ocaml-packages.nix [hunk 6]
```diff
@@ -1001,6 +1001,8 @@ let
 
     ppx_cstruct = callPackage ../development/ocaml-modules/cstruct/ppx.nix { };
 
+    ppx_cstubs = callPackage ../development/ocaml-modules/ppx_cstubs {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/ocaml-packages.nix:1004]:
```suggestion
    ppx_cstubs = callPackage ../development/ocaml-modules/ppx_cstubs { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by osener** [comment]
@SuperSandro2000 Thanks for the review! I've applied your suggestions.

**Review by vbgl — APPROVED** [review]

**Review by SuperSandro2000 — COMMENTED** [review]
diff LGTM, not sure if it is ocaml wise LGTM

**Pull request merged**